Eclipse article

danthe1st opened the pull request:
## Request issue
closes #68 

## Website section
Learn > Tutorials > Your First Java App

## Description
This article is an introductory tutorial for getting people started with developing Java applications in the Eclipse IDE.

There are no instructions about installing a JDK in this article because Eclipse should allow developing Java application with the JRE that comes with Eclipse.

## Notes
This article reflects my opinion of what such an article should look like. However, I am open to any feedback. This affects things like the structure or what I have (not) included.

Here are some specific decisions I have made for now that might be worth discussing/reconsidering:
- ~~Currently, it is directly in the `Your First Java App` section together with the VSC and upcoming IntelliJ article. It might be a good idea to create a subsection about IDEs and group them together. However, this should probably be done in synchronization with the IntelliJ article (#70) and I will await specific decisions from @ammbra.~~ The IntelliJ article was merged so I guess this will be done independently/afterwards.
- I have used https://www.eclipse.org/downloads/packages/installer instead of https://eclipseide.org for downloading Eclipse. I felt this was a more direct link to get it. Are there any reasons for having a link to <https://eclipseide.org>?
  - Eclipse also supports a portable installation without using the installer. I did not mention this type of installation.
- ~~The term "Eclipse" is a bit difficult as it shadows the differentiation between the Eclipse IDE and the Eclipse Foundation. Technically, the IDE is (as far as I know) called "Eclipse IDE" however most people just say "Eclipse". I used the term "Eclipse IDE" in the introduction but then switched to the shorter version as an attempt to both be correct but also not unnecessarily using the long (full) name.~~ I got confirmation that this should be fine.
- ~~I ended some lines with two spaces in order to add a line break before images. In most cases, this doesn't make a difference but there is at least one image where it would look weird otherwise (This may be different on other devices)~~ This was changed to use paragraphs.
![image](https://github.com/java/devjava-content/assets/34687786/c409e5cf-a828-4c44-83f3-7366c9c43b16)
- By default, Eclipse doesn't display packages hierarchically. I didn't include any information on how to change this as I also didn't write much about packages anyways.
- Similarly, Content Assist doesn't automatically activate on most keys. I just mentioned how to use it with a key combination. While it is possible to change this, I did not go over any settings/configuration in this article.
  - This key combination seems to be different on chinese systems (introduced [here](https://bugs.eclipse.org/bugs/show_bug.cgi?id=91327)) where seems to use the `Alt`-key. As MacOS normally uses `⌘` instead of `Alt`, I have included the corresponding key combination for chinese MacOS systems. However, I neither have a chinese nor a MacOS device so I wasn't able to test it (and I haven't found any references on the Content Assist keybind on chinese systems). See also <https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.platform.doc.user%2Fconcepts%2Faccessibility%2Fkeyboardshortcuts.htm>
- I have not yet found anything about Debugging keybinds on MacOS. I think these are probably the same (`F5`, `F6`, `F8`) ~~but I haven't checked that yet~~ it seems to be the same. It may be necessary to use the `Fn` key there.
- I generally used the UI navigation instead of mentioning keybindings as I felt this would be more stable. Maybe the summary section should contain something saying that most important actions can also be performed with keybindings that are often displayed next to the UI option?
- I don't know the article ID of the [records article](https://dev.java/learn/records/) as this is not available in this repository. Hence I used a `/learn/records` link.

## Not included
- Eclipse is Workspace-based (meaning one has a workspace with an arbitrary number of projects). I did not (yet) include any information about what that means and how it is different from Project-based IDEs.
- Perspectives and views are a fundamental concept in Eclipse which I was mentioning in the Debugging/Running sections. I plan to add a short section about perspectives/views as well.
- It might also be a good idea to add information about creating projects with Maven/Gradle in Eclipse. However, this would require informing users about setting the Java version in Maven project to not have a Java 5 (or 6 or whatever Maven's default is) project. I am not talking about a tutorial for using build tools but a section telling people how to create projects with build tools.
- I did not include any information on Testing (this is present in the IJ article). I don't think this is necessary for people writing their first applications.
- ~~Should there be a summary?~~
- I did not include anything about key combinations being shown in the perspective when no editor is opened.
- I have not included any links to documentation or `More Learning` resources. If it is helpful to include documentation or include a video for further learning, I am open for that.

danthe1st commented:
I found https://eclipseide.org/getting-started/documents/eclipse-ide-keybindings.pdf for keybindings - Debugging keybinds seem to be the same on MacOS but Content assist seems to be different.

danthe1st commented:
I now added sections for generating code, refactoring and I also added a short summary with a link to the User Guide.

tivervac commented on app/pages/learn/01_tutorial/01_your-first-java-app/03_writing-java-applications-with-eclipse.md:
```diff
@@ -0,0 +1,294 @@
+<a id="summary">&nbsp;</a>
+## Summary
+
+As you can see, the Eclipse IDE provides a lot of tools that help developers writing Java applications. While this article shows some, Eclipse comes with many more features which can be especially useful when working on bigger applications. If you are interested in reading more, check out the [Java Development user guide](https://help.eclipse.org/latest/index.jsp?nav=%2F1).
```
```suggestion
As you can see, the Eclipse IDE provides a lot of tools that help developers write Java applications. While this article shows some, Eclipse comes with many more features that can be especially useful when working on bigger applications. If you are interested in reading more, check out the [Java Development user guide](https://help.eclipse.org/latest/index.jsp?nav=%2F1).
```

carimura commented:
Hi @danthe1st, thanks again for the contribution. I'm back and can assist getting this over the line. I've started reviewing and pushing some fixes directly for expediency sake, hope that's OK. Mostly formatting, spacing, etc., but I did re-order slightly by moving the "Run Program" section above the "Dealing with errors" section. Let me know if that messes up your flow.

I've got a bit more review to do but will get back shortly.

danthe1st commented:
I have now added paragraph pages consistently and removed whitespace at the end of lines (which I used for line breaks previously).
Furthermore, I changed the file name to match the other articles.

carimura commented:
Thanks @tivervac and @BeckerWdf for the suggestions! We included a few of the trivial ones.

@danthe1st I'll get this merged!

carimura reviewed (approved):
LGTM

carimura commented:
We'll get this merged to prod within a day or so, thanks all!
